controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1527,6 +1527,17 @@ func (r *BareMetalHostReconciler) manageHostPower(prov provisioner.Provisioner,
 	// FIXME(janders/dtantsur) it would be preferrable to pass in state as an argument
 	// however this falls outside the scope of this specific change.
 
+	// If DataImage exists, handle attachment/detachment
+	handleDataImage := isProvisioned && desiredPowerOnState && !info.host.Status.PoweredOn
```

Comment:
This particular combination of flags is named as `servicingAllowed` on line 1542. Perhaps we could move that definition up here and reuse it.

Reply:
Makes sense, I will submit a pr to handle this.